### PR TITLE
Spec pricing / scopes / etc

### PR DESCRIPTION
**Pull request by shkm**
This:

- Gets specs running again. Needed to make some alterations for this
- Adds a bunch of stuff to common (see commits, I tried to keep it minimal)
- Specs out price determination
- Specs out scopes

### CODE REVIEW

File: config/initializers/common_initializer.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      trial_days: 0,
+      plan_name:  'affiliate',
+      name:       'Affiliate',
+      terms:      'Affiliate terms',
+    },
+    {
+      price:      10.0,
+      trial_days: 7,
+      name:       'Generic with trial',
+      terms:      'Generic germs',
```
Comment (mib32) [inline config/initializers/common_initializer.rb]: @shkm generic germs? 😂

Why is that file here and not in the dummy app?

File: lib/pr/common/models/user.rb [hunk 2]
```diff
@@ -7,7 +7,10 @@ module Models
       # So please include PR::Common::Models::User
       module User
         extend ActiveSupport::Concern
+
         included do
+          enum provider: { shopify: 0, tictail: 1 }
```
Comment (mib32) [inline lib/pr/common/models/user.rb]: @shkm Interested what @daniel-sim will say here

File: spec/lib/pr/common/shopify_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  describe '#determine_price' do
+    context 'shop has a plan whose pricing is defined' do
+      before { shop.update!(plan_name: 'affiliate') }
+
+      it 'returns the defined price' do
+        expected_price = PR::Common.config.pricing.detect do |pricing_plan|
+          pricing_plan[:plan_name] == 'affiliate'
+        end
+
+        expect(service.determine_price).to eq expected_price
```
Comment (mib32) [inline spec/lib/pr/common/shopify_service_spec.rb]: @shkm I in such cases prefer to use hard-coded (yeah! why not!) real values instead of expected_price just because who knows what is inside the expected_price? Maybe you mistaken and it is nil?

File: spec/lib/pr/common/shopify_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+describe PR::Common::ShopifyService do
+  let(:shop) { create(:shop, user: build(:user)) }
+  subject(:service) { described_class.new(shop: shop) }
+
+  describe '#determine_price' do
+    context 'shop has a plan whose pricing is defined' do
```
Comment (mib32) [inline spec/lib/pr/common/shopify_service_spec.rb]: @shkm What about the context 'shop has a plan whose pricing is undefined'?

File: spec/models/shop_spec.rb [hunk 5]
```diff
@@ -1,10 +1,25 @@
...
+
+  describe '.with_active_plan' do
+    it 'includes only shops whose plan is not cancelled' do
+      active_plan_shop = create(:shop)
+      cancelled_plan_shop = create(:shop, plan_name: 'cancelled')
+
+      shops = Shop.with_active_plan
+      expect(shops).to include active_plan_shop
+      expect(shops).not_to include cancelled_plan_shop
+    end
```
Comment (mib32) [inline spec/models/shop_spec.rb]: @shkm Forgot to describe '.installed'

File: config/initializers/common_initializer.rb (hunk 1 above)
Comment (shkm) [reply config/initializers/common_initializer.rb]: Oops! Haha.

And yeah, good catch. It's meant to be in the dummy app.

File: lib/pr/common/models/user.rb (hunk 2 above)
Comment (shkm) [reply lib/pr/common/models/user.rb]: Indeed, me too. Leave this open until he comments.

File: spec/lib/pr/common/shopify_service_spec.rb (hunk 3 above)
Comment (shkm) [reply spec/lib/pr/common/shopify_service_spec.rb]: Yeah, I'm never too sure about these either, so I'm liable to agree with you.

File: spec/lib/pr/common/shopify_service_spec.rb (hunk 4 above)
Comment (shkm) [reply spec/lib/pr/common/shopify_service_spec.rb]: Yeah that's worth testing. Will add.

File: spec/models/shop_spec.rb (hunk 5 above)
Comment (shkm) [reply spec/models/shop_spec.rb]: True!

File: config/initializers/common_initializer.rb (hunk 1 above)
Comment (shkm) [reply config/initializers/common_initializer.rb]: ✅

File: spec/lib/pr/common/shopify_service_spec.rb (hunk 3 above)
Comment (shkm) [reply spec/lib/pr/common/shopify_service_spec.rb]: ✅

File: spec/lib/pr/common/shopify_service_spec.rb (hunk 4 above)
Comment (shkm) [reply spec/lib/pr/common/shopify_service_spec.rb]: :check

File: spec/models/shop_spec.rb (hunk 5 above)
Comment (shkm) [reply spec/models/shop_spec.rb]: ✅

**Comment by shkm** [comment]
OK, everything fixed apart from the enum.
